### PR TITLE
Bug 1829959: [release-4.3] Introduce IBMCloud provider, skip/fix tests

### PR DESCRIPTION
**Pull request by csrwng**
Backport of #24817

### CODE REVIEW

File: test/extended/util/test.go [hunk 1]
```diff
@@ -484,6 +485,55 @@ var (
...
+
+			// TestFrontProxy relies on the aggregator client secret to be present on the cluster.
+			// On a hosted control plane cluster, that secret lives on the management side.
+			`\[Feature:Authentication\]  TestFrontProxy should succeed`,
+
+			// Requires the kube-control-plane-signer secret which is not present in a ROKS cluster
+			`\[Feature:OAuthServer\] OAuth server has the correct token and certificate fallback semantics`,
+
+			// Requires oauth-openshift route in cluster. In ROKS, the oauth server lives outside of the cluster
+			`\[Feature:OAuthServer\] well-known endpoint should be reachable`,
```
Comment (derekwaynecarr) [inline test/extended/util/test.go:505]: fix this and not skip it.

File: test/extended/util/test.go [hunk 2]
```diff
@@ -484,6 +485,55 @@ var (
...
+			"The bootstrap user should successfully login with password decoded from kubeadmin secret",
+
+			// TestFrontProxy relies on the aggregator client secret to be present on the cluster.
+			// On a hosted control plane cluster, that secret lives on the management side.
+			`\[Feature:Authentication\]  TestFrontProxy should succeed`,
+
+			// Requires the kube-control-plane-signer secret which is not present in a ROKS cluster
+			`\[Feature:OAuthServer\] OAuth server has the correct token and certificate fallback semantics`,
+
+			// Requires oauth-openshift route in cluster. In ROKS, the oauth server lives outside of the cluster
```
Comment (csrwng) [inline test/extended/util/test.go:504]: Should fix test to use alternate endpoint vs route

File: test/extended/util/test.go [hunk 3]
```diff
@@ -484,6 +485,55 @@ var (
...
+			`\[sig-auth\] \[Feature:NodeAuthorizer\] Getting a secret for a workload the node has access to should succeed`,
+			`\[sig-auth\] \[Feature:NodeAuthorizer\] Getting an existing configmap should exit with the Forbidden error`,
+			`\[sig-auth\] \[Feature:NodeAuthorizer\] Getting an existing secret should exit with the Forbidden error`,
+
+			// Access to node external address is blocked from pods within a ROKS cluster by Calico
+			`\[sig-auth\] \[Feature:NodeAuthenticator\] The kubelet's main port 10250 should reject requests with no credentials`,
+			`\[sig-auth\] \[Feature:NodeAuthenticator\] The kubelet can delegate ServiceAccount tokens to the API server`,
+
+			// Node approval in ROKS is not handled in cluster
+			`node client cert requests armoring: node-approver SA token compromised, don't approve random CSRs with client auth`,
```
Comment (csrwng) [inline test/extended/util/test.go:519]: Add the skip in the test based on platform

File: test/extended/util/test.go [hunk 4]
```diff
@@ -484,6 +485,55 @@ var (
...
+			`\[Feature:OAuthServer\] well-known endpoint should be reachable`,
+
+			// Nodes in ROKS have access to secrets in the cluster to handle encryption
+			`\[sig-auth\] \[Feature:NodeAuthorizer\] Getting a non-existent configmap should exit with the Forbidden error, not a NotFound error`,
+			`\[sig-auth\] \[Feature:NodeAuthorizer\] Getting a non-existent secret should exit with the Forbidden error, not a NotFound error`,
+			`\[sig-auth\] \[Feature:NodeAuthorizer\] Getting a secret for a workload the node has access to should succeed`,
+			`\[sig-auth\] \[Feature:NodeAuthorizer\] Getting an existing configmap should exit with the Forbidden error`,
+			`\[sig-auth\] \[Feature:NodeAuthorizer\] Getting an existing secret should exit with the Forbidden error`,
+
+			// Access to node external address is blocked from pods within a ROKS cluster by Calico
```
Comment (csrwng) [inline test/extended/util/test.go:514]: Should have a bug on our side to go fix upstream

File: test/extended/util/test.go [hunk 5]
```diff
@@ -484,6 +485,55 @@ var (
...
+
+			// Ignition is not served from a ROKS cluster
+			`node client cert requests armoring: deny pod's access to /config/master API endpoint`,
+
+			// No metrics are available for control plane components
+			`\[Feature:Prometheus\]\[Conformance\] Prometheus when installed on the cluster should start and expose a secured proxy and unsecured metrics`,
+
+			// The cluster-network-operator creates the kube-proxy daemonset pods without mem/cpu requests,
+			// resulting in a qosClass of BestEffort
+			`\[Feature:Platform\] Managed cluster should ensure control plane pods do not run in best-effort QoS`,
```
Comment (derekwaynecarr) [inline test/extended/util/test.go:529]: set to 1/2 of openshift-sdn and fix in our code.

Comment (csrwng) [inline test/extended/util/test.go:529]: Open bug for network-cluster-operator.

Comment (csrwng) [reply test/extended/util/test.go:529]: Set requests to half of what openshift-sdn is using

**Comment by csrwng** [comment]
/test e2e-gcp
Error creating cluster

**Pull request merged**